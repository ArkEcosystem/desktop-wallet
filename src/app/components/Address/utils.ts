/**
 * Truncates a given string adding
 * dots in the middle (e.g AUdQs...SfdQs).
 */
export const truncateStringMiddle = (input: string, maxChars = 20) => {
	if (input.length <= maxChars) return input;

	const midPos = Math.floor(maxChars / 2) - 2;
	const start = input.substr(0, midPos);
	const end = input.substr(input.length - midPos, input.length);
	return `${start}...${end}`;
};
