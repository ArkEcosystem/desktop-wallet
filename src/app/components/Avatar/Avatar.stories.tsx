import React from "react";

import { Avatar } from "./Avatar";

export default { title: "App / Components / Avatar" };

export const Default = () => {
	return (
		<div className="inline-flex space-x-5">
			<Avatar address="APTz" size="small" />
			<Avatar address="AaAy" size="default" />
			<Avatar address="1Pdj" size="large" />
		</div>
	);
};
