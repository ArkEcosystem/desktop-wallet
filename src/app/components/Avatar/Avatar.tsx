import { Avatar as AvatarSDK } from "@arkecosystem/platform-sdk-profiles";
import React from "react";
import tw, { css, styled } from "twin.macro";
import { Size } from "types";

type Props = {
	address: string;
	size?: Size;
	noShadow?: boolean;
	className?: string;
	shadowColor?: string;
};

export const AvatarWrapper = styled.div<{ shadowColor?: string; size?: string; noShadow?: boolean }>`
	${tw`block rounded-full overflow-hidden`}
	${({ size }) => {
		switch (size) {
			case "sm":
				return tw`w-8 h-8`;
			case "lg":
				return tw`w-12 h-12`;
			default:
				return tw`w-10 h-10`;
		}
	}}
	${({ noShadow, shadowColor }) =>
		!noShadow &&
		css`
			& {
				box-shadow: 0 0 0 6px var(${shadowColor ? shadowColor : "--theme-background-color"});
			}
		`};
`;

export const Avatar = ({ address, size, noShadow, className, shadowColor }: Props) => {
	const svg = React.useMemo(() => AvatarSDK.make(address), [address]);

	return (
		<AvatarWrapper
			data-testid="Avatar"
			size={size}
			noShadow={!!noShadow}
			className={className}
			shadowColor={shadowColor}
		>
			<img alt={address} title={address} src={`data:image/svg+xml;utf8,${svg}`} />
		</AvatarWrapper>
	);
};
