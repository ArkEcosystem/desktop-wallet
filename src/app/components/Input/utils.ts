type RangeInput = {
	min: number;
	max: number;
	step: number;
};

const isStepDivisible = ({ min, max, step }: RangeInput) => {
	const res = (max - min) / step;
	return parseInt(res.toString(), 10) === res;
};

export const sanitizeStep = ({ min, max, step }: RangeInput): number =>
	isStepDivisible({ min, max, step }) ? step : (max - min) / 100;
