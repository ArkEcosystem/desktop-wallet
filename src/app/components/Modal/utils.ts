export const modalOffsetClass = (itemHeight: number, windowHeight: number) => {
	const padding = 80;

	return itemHeight + 2 * padding > windowHeight ? "top-0 my-20" : "";
};
