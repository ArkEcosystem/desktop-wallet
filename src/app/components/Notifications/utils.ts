import { Profile } from "@arkecosystem/platform-sdk-profiles";

export const markAsRead = async (isVisible: boolean, id: string, profile: Profile, env: any) => {
	if (!isVisible) {
		return;
	}

	const notification = profile.notifications().get(id);
	if (!notification || notification?.read_at) {
		return;
	}

	profile.notifications().markAsRead(id);
	await env.persist();
};
