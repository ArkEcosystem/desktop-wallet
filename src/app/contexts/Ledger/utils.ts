import { BigNumber } from "@arkecosystem/platform-sdk-support";

export type LedgerData = {
	index: number;
	address: string;
	path: string;
	balance?: BigNumber;
	isNew?: boolean;
	timestamp?: number;
};

export type LedgerDerivationScheme = {
	coinType: number;
	purpose?: number;
	account?: number;
	change?: number;
	address?: number;
};

export const customDerivationModes = {
	ARK: {
		legacy: (slip44: number, cursor: number) => formatLedgerDerivationPath({ coinType: slip44, account: cursor }),
		bip44: (slip44: number, cursor: number) => {
			/** The legacy mode also starts with this path, so it doesn't need to be scanned again */
			if (cursor === 0) {
				return;
			}

			return formatLedgerDerivationPath({ coinType: slip44, address: cursor });
		},
	},
};

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${scheme.address || 0}`;
