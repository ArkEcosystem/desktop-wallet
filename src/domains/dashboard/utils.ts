import { Profile, ProfileSetting } from "@arkecosystem/platform-sdk-profiles";
import { uniq } from "@arkecosystem/utils";

export const enableNetworkInDashboardFilters = (profile: Profile, networkId: string) => {
	const config: any = profile.settings().get(ProfileSetting.DashboardConfiguration) || { selectedNetworkIds: [] };
	config.selectedNetworkIds.push(networkId);
	profile.settings().set(ProfileSetting.DashboardConfiguration, {
		...config,
		selectedNetworkIds: uniq(config.selectedNetworkIds),
	});
};
