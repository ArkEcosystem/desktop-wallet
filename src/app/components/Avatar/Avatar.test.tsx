import React from "react";
import { render } from "utils/testing-library";

import { Avatar } from "./Avatar";

describe("Avatar", () => {
	it("should render", () => {
		const { getByTestId, asFragment } = render(<Avatar address="abc" />);
		expect(getByTestId("Avatar")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with highlight", () => {
		const { getByTestId, asFragment } = render(<Avatar address="abc" highlight />);
		expect(getByTestId("Avatar")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with custom shadow color", () => {
		const { getByTestId, asFragment } = render(<Avatar address="abc" shadowColor="--theme-background-color" />);
		expect(getByTestId("Avatar")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with highlight and custom shadow color", () => {
		const { getByTestId, asFragment } = render(
			<Avatar address="abc" shadowColor="--theme-background-color" highlight />,
		);
		expect(getByTestId("Avatar")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render without shadow", () => {
		const { getByTestId, asFragment } = render(<Avatar address="abc" size="lg" noShadow />);
		expect(getByTestId("Avatar")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it.each(["sm", "lg", "xl"])("should render with size", (size) => {
		const { getByTestId, asFragment } = render(<Avatar address="abc" size={size} />);
		expect(getByTestId("Avatar")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});
});
