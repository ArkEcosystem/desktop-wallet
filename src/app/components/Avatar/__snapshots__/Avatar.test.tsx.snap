// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render 1`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 gYYjOs"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar should render with custom shadow color 1`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 gYYjOs"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar should render with highlight 1`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 jDqCgJ"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar should render with highlight and custom shadow color 1`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 jDqCgJ"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar should render with size 1`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 keFjxD"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar should render with size 2`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 gyYbkb"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar should render with size 3`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 cPTwWa"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar should render without shadow 1`] = `
<DocumentFragment>
  <div
    class="Avatar__AvatarWrapper-sc-1vsy2s2-0 cjwvlc"
    data-testid="Avatar"
  >
    <div
      class="w-full h-full"
    >
      <img
        alt="abc"
        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(76, 175, 80)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"30\\" cy=\\"60\\" fill=\\"rgb(255, 193, 7)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"60\\" cx=\\"80\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/></svg>"
        title="abc"
      />
    </div>
  </div>
</DocumentFragment>
`;
